src/box/box.cc

```diff
@@ -4555,14 +4571,11 @@ box_process_join(struct iostream *io, const struct xrow_header *header)
 	 * Register the replica after sending the last row but before sending
 	 * OK - if the registration fails, the error reaches the client.
 	 */
-	box_register_replica(&req.instance_uuid, req.instance_name);
+	replica = box_register_replica_with_consumer(
+		&req.instance_uuid, req.instance_name, &gc->vclock);
```

Comment:
> there'll be several consumers with the same name

In-memory consumer has name `replica ... join` unlike an usual one :)
But anyway, there is no need to keep it after the registration, so I unregister it right after registration now.

Reply:
We should create consumer only after registering replica in `_cluster`, otherwise the consumer may become hanging if the join will fail before the registration and replica won't return, so we can't do `_gc_consumer:replace` in `relay_initial_join`. Or we could register replica before initial join, but I'm not sure it's a good idea.

What's about in-memory consumer, it should be unregistered strictly after registration because otherwise required xlogs will be most-likely deleted when the fiber will yield on replica registration.